profiling/src/profiling.rs

```diff
@@ -187,29 +180,22 @@ unsafe fn extract_function_name(func: &zend_function) -> String {
     let mut buffer = Vec::<u8>::new();
 
     // User functions do not have a "module". Maybe one day use composer info?
-    if func.type_ == ZEND_INTERNAL_FUNCTION as u8
-        && !func.internal_function.module.is_null()
-        && !(*func.internal_function.module).name.is_null()
-    {
-        let ptr = (*func.internal_function.module).name as *const c_char;
-        let bytes = CStr::from_ptr(ptr).to_bytes();
-        if !bytes.is_empty() {
-            buffer.extend_from_slice(bytes);
-            buffer.push(b'|');
-        }
+    let module_name = func.module_name().unwrap_or(b"");
+    if !module_name.is_empty() {
+        buffer.extend_from_slice(module_name);
+        buffer.push(b'|');
     }
 
-    if !func.common.scope.is_null() {
-        let class_name = zend_string_to_bytes((*func.common.scope).name);
-        if !class_name.is_empty() {
-            buffer.extend_from_slice(class_name);
-            buffer.extend_from_slice(b"::");
-        }
+    // todo: probably use EX(This) instead of func.common.scope.
```

Comment:
I think using the func.common.scope is better here, given that stacktraces generally mirror the declaration context for methods and not the run-time late static binding or $this context.